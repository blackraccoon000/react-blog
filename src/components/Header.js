import React from "react"
import NavMenu from "./NavMenu"
import { connect } from "react-redux"
import NoteListFilters from "./NoteListFilters"
class Header extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      flag:false
    }
  }
  onClickBtn = () => {
    this.setState({flag: !this.state.flag })
  }
  render() {
    return (
      <div>
        <header className="header">
          <div className="content-container">
            <h1 className="header__title">My Blog</h1>
            { this.state.flag && <NavMenu history={history}/> }
            { this.state.flag && <NoteListFilters/> }
            <button className="navBtn" data-testid="navBarFlagButton" onClick={this.onClickBtn}>
              {this.state.flag ? "Close" : "Open" }
            </button>
          </div>
        </header>
        <div className="header-alert">
          <div className="content-container">
            { this.props.isAuthenticated && "sign in now" }
          </div>
        </div>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  isAuthenticated: !!state.auth.uid
})

export { Header }
export default connect( mapStateToProps )( Header )