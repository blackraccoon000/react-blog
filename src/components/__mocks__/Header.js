/**
 * Headerは執筆時クラス化してある。
 * mock化にあたり関数化してあるので、そのつもりで。
 */

import React from "react"
const mock = jest.fn().mockImplementation(() => (<div>HeaderPage Use ManualMock</div>))
export default mock