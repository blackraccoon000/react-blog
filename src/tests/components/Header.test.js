import React from "react"
import { shallow, render } from "enzyme"
import Header from "../../components/Header"

let wrapper, onClickBtn

beforeEach(()=>{
  onClickBtn = jest.fn()
  wrapper = shallow(<Header onClickBtn={onClickBtn}/>)
})

it("should render Header Components", () => {
  expect(wrapper).toMatchSnapshot()
})

// it("should Header Components Button Click for flag false => true", () => {
//   expect(wrapper.find("button").length).toBe(1)
//   expect(wrapper.state("flag")).toBe(false)
//   wrapper.find("button").simulate("click")
//   expect(wrapper.state("flag")).toBe(true)
//   expect(wrapper.find("NavMenu").length).toBe(1)
//   expect(wrapper).toMatchSnapshot()
// })