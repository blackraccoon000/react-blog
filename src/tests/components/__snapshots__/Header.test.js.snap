// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header Components 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <h1
      className="header__title"
    >
      My Blog
    </h1>
    <button
      onClick={[Function]}
    >
      open
    </button>
  </div>
</header>
`;
